src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/IconUrlToImageCacheConverter.cs

```diff
@@ -54,8 +61,62 @@ public object Convert(object value, Type targetType, object parameter, CultureIn
 
             var iconBitmapImage = new BitmapImage();
             iconBitmapImage.BeginInit();
-            iconBitmapImage.UriSource = iconUrl;
 
+            BitmapSource imageResult;
+
+            if (values.Length > 1
```

Comment:
Why is the 2nd value so important? Can you put a brief comment here that explains?

Reply:
The second value is the `PackageArchiveReader` from which we read the icon. I added comments and some if statements

Reply:
Sure. Added documentation to the xmldoc for Convert()